src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2891,7 +2889,7 @@ public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(Dyn
                 {
                     if (returnType is null)
                         // if we are here we have no args or property type to select over and we have more than one property with that name
-                        throw new AmbiguousMatchException();
+                        ThrowHelper.ThrowAmbiguousMatchException(firstCandidate);
```

Comment:
As this is the first use of the helper in this PR,  I thought I would mention the strategy. To resolve the original issue and add more information to figure out _what_ was ambiguous, we need to know the name of the member that has ambiguity. 

Since `PropertyInfo` et al all descend from `MemberInfo`, we can make a helper that formats messages from calling `.ToString()` on the info and it's `DeclaringType`. This yields, for example from `String.Length` the values `System.String` and `Int32 Length` as the insertions. See _ThrowHelper.cs_